### ISSUE TITLE
WebSockets custom function roles crash

### ISSUE THREAD

**Issue by chris-feist** [serverless/serverless#5868]
# This is a Bug Report

## Description

* What went wrong?

When using custom function roles and wesockets, Serverless crashes. 

Example serverless.yml:
```yaml
functions:
  myWebSocket:
    handler: myWebsocket.handler
    role: { "Fn::GetAtt": [ "MyWebSocketLambdaFunctionRole", "Arn" ] }
    events:
      - websocket:
          route: $connect
      - websocket:
          route: $disconnect
      - websocket:
          route: $default

resources:
  Resources:
    MyWebSocketLambdaFunctionRole: 
      Type: "AWS::IAM::Role,
      Properties:
        #custom role policies
```

Stack trace:

* What did you expect should have happened?
 No crashes and allow my custom roles

* What was the config you used?
* What stacktrace or error message from your provider did you see?
```bash
  Type Error ---------------------------------------------
 
  Cannot read property 'Properties' of undefined
 
     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.
 
  Stack Trace --------------------------------------------
 
TypeError: Cannot read property 'Properties' of undefined
    at AwsCompileWebsockets.compileApi (/Users/...project/node_modules/serverless/lib/plugins/aws/package/compile/events/websockets/lib/api.js:39:8)
From previous event:
    at Object.package:compileEvents [as hook] (/Users/...project/node_modules/serverless/lib/plugins/aws/package/compile/events/websockets/index.js:39:12)
    at BbPromise.reduce (/Users/...project/node_modules/serverless/lib/classes/PluginManager.js:407:55)
From previous event:
    at PluginManager.invoke (/Users/...project/node_modules/serverless/lib/classes/PluginManager.js:407:22)
    at PluginManager.spawn (/Users/...project/node_modules/serverless/lib/classes/PluginManager.js:425:17)
    at Deploy.BbPromise.bind.then (/Users/...project/node_modules/serverless/lib/plugins/deploy/deploy.js:117:50)
From previous event:
    at Object.before:deploy:deploy [as hook] (/Users/...project/node_modules/serverless/lib/plugins/deploy/deploy.js:107:10)
    at BbPromise.reduce (/Users/...project/node_modules/serverless/lib/classes/PluginManager.js:407:55)
From previous event:
    at PluginManager.invoke (/Users/...project/node_modules/serverless/lib/classes/PluginManager.js:407:22)
    at PluginManager.run (/Users/...project/node_modules/serverless/lib/classes/PluginManager.js:438:17)
    at variables.populateService.then.then (/Users/...project/node_modules/serverless/lib/Serverless.js:114:33)
    at processImmediate (timers.js:637:19)
    at process.topLevelDomainCallback (domain.js:120:23)
From previous event:
    at Serverless.run (/Users/...project/node_modules/serverless/lib/Serverless.js:101:6)
    at serverless.init.then (/Users/...project/node_modules/serverless/bin/serverless:43:28)
    at /Users/...project/node_modules/graceful-fs/graceful-fs.js:111:16
    at /Users/...project/node_modules/graceful-fs/graceful-fs.js:45:10
    at FSReqCallback.args [as oncomplete] (fs.js:146:20)
From previous event:
    at initializeErrorReporter.then (/Users/...project/node_modules/serverless/bin/serverless:43:6)
    at processImmediate (timers.js:637:19)
    at process.topLevelDomainCallback (domain.js:120:23)
From previous event:
    at /Users/...project/node_modules/serverless/bin/serverless:28:46
    at Object.<anonymous> (/Users/...project/node_modules/serverless/bin/serverless:67:4)
    at Module._compile (internal/modules/cjs/loader.js:734:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:745:10)
    at Module.load (internal/modules/cjs/loader.js:626:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:566:12)
    at Function.Module._load (internal/modules/cjs/loader.js:558:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:797:12)
    at executeUserCode (internal/bootstrap/node.js:526:15)
    at startMainThreadExecution (internal/bootstrap/node.js:439:3)
```

Similar or dependent issues:
* #5824 

## Additional Data

* ***Serverless Framework Version you're using***: 1.38.0
* ***Operating System***: macOS 10.14
* ***Stack Trace***:
* ***Provider Error messages***:

**Comment by eahefnawy** [comment]
Thanks a lot for reporting this @chris-feist . I think I know what's going on, I'll push a fix very soon.

**Comment by eahefnawy** [comment]
@chris-feist I've pushed a fix for this in #5880 ... could you try it out and confirm that it fixes your issue?

**Comment by chris-feist** [comment]
@eahefnawy I can confirm that fixed the issue. Thanks for the quick turnaround 👍
